api/consensus.go

```diff
@@ -19,6 +21,55 @@ type ConsensusGET struct {
 	Difficulty   types.Currency    `json:"difficulty"`
 }
 
+type ConsensusFileContract struct {
+	FileSize           uint64                         `json:"filesize"`
+	FileMerkleRoot     crypto.Hash                    `json:"filemerkleroot"`
+	WindowStart        types.BlockHeight              `json:"windowstart"`
+	WindowEnd          types.BlockHeight              `json:"windowend"`
+	Payout             types.Currency                 `json:"payout"`
+	ValidProofOutputs  map[string]types.SiacoinOutput `json:"validproofoutputs"`
+	MissedProofOutputs map[string]types.SiacoinOutput `json:"missedproofoutputs"`
+	UnlockHash         types.UnlockHash               `json:"unlockhash"`
+	RevisionNumber     uint64                         `json:"revisionnumber"`
+}
+
+type ConsensusFileContractRevision struct {
+	ParentID          types.FileContractID   `json:"parentid"`
+	UnlockConditions  types.UnlockConditions `json:"unlockconditions"`
+	NewRevisionNumber uint64                 `json:"newrevisionnumber"`
+
+	NewFileSize           uint64                         `json:"newfilesize"`
+	NewFileMerkleRoot     crypto.Hash                    `json:"newfilemerkleroot"`
+	NewWindowStart        types.BlockHeight              `json:"newwindowstart"`
+	NewWindowEnd          types.BlockHeight              `json:"newwindowend"`
+	NewValidProofOutputs  map[string]types.SiacoinOutput `json:"newvalidproofoutputs"`
+	NewMissedProofOutputs map[string]types.SiacoinOutput `json:"newmissedproofoutputs"`
+	NewUnlockHash         types.UnlockHash               `json:"newunlockhash"`
+}
+
+type ConsensusTransaction struct {
+	SiacoinInputs         map[string]types.SiacoinInput            `json:"siacoininputs"`
+	SiacoinOutputs        map[string]types.SiacoinOutput           `json:"siacoinoutputs"`
+	FileContracts         map[string]ConsensusFileContract         `json:"filecontracts"`
+	FileContractRevisions map[string]ConsensusFileContractRevision `json:"filecontractrevisions"`
```

Comment:
using custom structures with `map[string]` instead of types with `map[]` to fill it with any type hashes to make possibility index it with external tools.

Sia internally generate hash every time when it need, like this:
https://github.com/NebulousLabs/Sia/blob/master/types/transactions.go#L217-L248

So default structures doesn't contain hash of anything.

Check this out sia explorer vs my indexer: 
https://github.com/S-anasol/SiaConsensusIndexer/blob/master/update.php#L58-L119

https://github.com/NebulousLabs/Sia/blob/master/modules/explorer/update.go#L59

Sia Explorer uses internal functions to calculate hash(ID) for data array,
So making indexer in other language with full copy of Sia hashes calculation is complicated and overheaded.

I hope understandable explanation :)